src/spatio_temporal_voxel_grid.cpp

```diff
@@ -451,4 +451,52 @@ bool SpatioTemporalVoxelGrid::SaveGrid(
   return false;  // best offense is a good defense
 }
 
+/*****************************************************************************/
+void SpatioTemporalVoxelGrid::clearExceptRegion(
+  double robot_x, double robot_y, double reset_distance,
+  std::unordered_set<occupany_cell> & cleared_cells)
+/*****************************************************************************/
+{
+  boost::unique_lock<boost::mutex> lock(_grid_lock);
+
+  if (this->IsGridEmpty()) {
+    return;
+  }
+
+  _grid_points->clear();
+  _cost_map->clear();
+
+  openvdb::DoubleGrid::ValueOnCIter cit_grid = _grid->cbeginValueOn();
+  for (; cit_grid.test(); ++cit_grid) {
+
+    const openvdb::Coord pt_index(cit_grid.getCoord());
+    const openvdb::Vec3d pose_world = this->IndexToWorld(pt_index);
+    bool cleared_point = false;
+
+    double start_point_x = robot_x - reset_distance / 2;
+    double start_point_y = robot_y - reset_distance / 2;
+    double end_point_x = start_point_x + reset_distance;
+    double end_point_y = start_point_y + reset_distance;
+
+    double x = pose_world[0];
+    double y = pose_world[1];
+
+    bool xrange = x <= start_point_x || x >= end_point_x;
+    bool yrange = y <= start_point_y || y >= end_point_y;
+
+    if (xrange && yrange){
+      if (!this->ClearGridPoint(pt_index)) {
+        std::cout << "Failed to clear point." << std::endl;
+      }
+      else {
+        cleared_cells.insert(occupany_cell(x, y));
+      }
```

Comment:
```suggestion
      if (this->ClearGridPoint(pt_index)) {
        cleared_cells.insert(occupany_cell(x, y));
      }
      else {
        std::cout << "Failed to clear point." << std::endl;
      }
```
This might be slightly easier to read this way around, without negations. This is just mostly preference question...

Reply:
I agree, inverted it to look better!